src/components/WalletUnlock/HWLegacy.tsx

```diff
@@ -177,9 +203,16 @@ class LedgerNanoSDecryptClass extends PureComponent<Props & INetworkContext, Sta
     this.setState({
       isConnected: false,
       dPath:
-        getDPath(network, WalletId.LEDGER_NANO_S) || getDPaths(networks, WalletId.LEDGER_NANO_S)[0]
+        getDPath(network, this.props.wallet.id as DPathFormat) ||
+        getDPaths(networks, this.props.wallet.id as DPathFormat)[0]
     });
   }
 }
 
-export const LedgerNanoSDecrypt = withHook(useNetworks)(LedgerNanoSDecryptClass);
+const mapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
+  bindActionCreators({ connectWallet }, dispatch);
+
+const connector = connect(() => ({}), mapDispatchToProps);
+type Props = ConnectedProps<typeof connector> & OwnProps;
+
+export const HWLegacy = withHook(useNetworks)(connector(HWLegacyClass));
```

Comment:
Can you test this file?

Reply:
I've tested most of the file. The rest is not easily testable at this point.